src/plugins/intel_gpu/src/plugin/program.cpp

```diff
@@ -112,26 +144,136 @@ Program::Program(InferenceEngine::CNNNetwork& network, std::shared_ptr<cldnn::en
 
     auto ops = func->get_ordered_ops();
 
-    if (m_config.max_dynamic_batch > 1) {
-        // check topology for applicability
-        if (!CanProcessDynBatch(ops, networkInputs)) {
-            IE_THROW() << "Such topology cannot be compiled for dynamic batch!";
+    bool dyn_shape_batch_found = false;
+    std::map<std::string, ngraph::PartialShape> shapes;
+    std::map<std::string, std::pair<int64_t, int64_t>> batch_dim;
+    if (m_config.enableDynamicBatch) {
+        // in case of legacy dynamic batch,
+        // we assume 4D input with 0 batch dim
+        auto param = func->get_parameters().front();
+        auto pname = getParamName(param);
+        shapes[pname] = param->get_output_partial_shape(0);
+        batch_dim[pname].first = 0;
```

Comment:
Maybe will be better to ask batch dim from the model layout?

Reply:
It is impossible to retrieve batch dimension from the model layout, if it hasn't been set manually (model has empty dimension string)
Unfortunately, this is the case for all our auto-generated test model.
Attempt to rely on model layout immediately breaks CI tests (at least in my case).